packages/flutter_tools/lib/src/build_system/targets/macos.dart

```diff
@@ -83,6 +84,102 @@ class UnpackMacOS extends Target {
   }
 }
 
+/// Compile an App.framework for a macOS target device.
+class MacOSAotAssembly extends Target {
+  const MacOSAotAssembly();
+
+  @override
+  String get name => 'macos_aot_assembly';
+
+  @override
+  Future<void> build(List<File> inputFiles, Environment environment) async {
+    final AOTSnapshotter snapshotter = AOTSnapshotter(reportTimings: false);
+    final String outputPath = fs.path.join(environment.projectDir.path, 'macos', 'Flutter', 'ephemeral');
+    if (environment.defines[kBuildMode] == null) {
+      throw MissingDefineException(kBuildMode, 'macos_aot_assembly');
+    }
+    if (environment.defines[kTargetPlatform] == null) {
+      throw MissingDefineException(kTargetPlatform, 'macos_aot_assembly');
+    }
+    final BuildMode buildMode = getBuildModeForName(environment.defines[kBuildMode]);
+    final TargetPlatform targetPlatform = getTargetPlatformForName(environment.defines[kTargetPlatform]);
+    if (targetPlatform != TargetPlatform.darwin_x64) {
+      throw Exception('macos_aot_assembly is only supported for iOS application.s');
+    }
+    if (buildMode == BuildMode.debug) {
+      throw Exception('macos_aot_assembly is only supported in profile or release mode.');
+    }
+    final int snapshotExitCode = await snapshotter.build(
+      platform: targetPlatform,
+      buildMode: buildMode,
+      mainPath: environment.buildDir.childFile('app.dill').path,
+      packagesPath: environment.projectDir.childFile('.packages').path,
+      outputPath: outputPath,
+      iosArch: IOSArch.x86_64,
+      bitcode: false,
+    );
+    if (snapshotExitCode != 0) {
+      throw Exception('AOT snapshotter exited with code $snapshotExitCode');
+    }
+  }
+
+  @override
+  List<Target> get dependencies => const <Target>[
+    KernelSnapshot(),
+  ];
+
+  @override
+  List<Source> get inputs => const <Source>[
+    Source.pattern('{BUILD_DIR}/app.dill')
+  ];
+
+  @override
+  List<Source> get outputs => <Source>[
+    const Source.pattern('{PROJECT_DIR}/macos/Flutter/ephemeral/App.framework/App')
+  ];
+}
+
+/// Create an App.framework for debug targets.
+///
+/// This framework needs to exist for the Xcode project to link/bundle,
+/// but it isn't actually executed. To generate something valid, we compile a trivial
```

Comment:
Is this how the iOS App.framework works?

Reply:
Yes, it is an interesting approach:

https://github.com/flutter/flutter/blob/master/packages/flutter_tools/bin/xcode_backend.sh#L238